src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/routing/AsyncStoragePool.java

```diff
@@ -15,45 +15,32 @@
 
 package com.amazon.opendistroforelasticsearch.security.auditlog.routing;
 
+import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditMessage;
+import com.amazon.opendistroforelasticsearch.security.auditlog.sink.AuditLogSink;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-import org.elasticsearch.common.settings.Settings;
-
-import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditMessage;
-import com.amazon.opendistroforelasticsearch.security.auditlog.sink.AuditLogSink;
-import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
-
 public class AsyncStoragePool {
```

Comment:
Looks like AsyncStoragePool and AsyncStorageConfig are generally used together. Any reason for splitting the latter to a separate class.

Reply:
Its always a good idea to abstract out and bundle configuration from the service/class dependent on it instead of having configuration parameters fly around. 

This will simplify injecting configuration through constructor dependency injection or via hot-reloading. It also makes changes to configuration more maintainable. 

In this case, AsyncStoragePool is dependent on configuration AsyncStorageConfig